src/tts_model.cpp

```diff
@@ -15,6 +15,23 @@ void append_to_response(struct tts_response * response, struct tts_response * to
     response->n_outputs += to_append->n_outputs;
 }
 
+/* 
+ * Pulls output_size to prepped buffer 'output' from 'output_node' tensor. If no buffer is passed will default to the existing output buffer present 
+ * on runner_context. 
+ */
+void runner_context::get_ggml_node_data(struct ggml_tensor * output_node, float * output, const size_t output_size, ggml_backend_buffer_t buffer) {
+    if (buffer == nullptr) {
+        buffer = buf_output;
+    }
+    if (ggml_backend_buffer_get_size(buffer) < output_size) {
+        TTS_ABORT("Output buffer overflow of %d / %d for output node '%s'\n", output_size, ggml_backend_buffer_get_size(buffer), ggml_get_name(output_node));
+    } else if (ggml_nbytes(output_node) < output_size) {
```

Comment:
This second `<` can be a `!=` for a stronger assertion if you want

Reply:
I thought about this, but It is a common pattern to only grab the first or last element in a series tensor (in order to avoid adding unnessary or arbitrary view op).